source/Module/STIG/Functions.Checklist.ps1

```diff
@@ -240,18 +240,36 @@ function New-StigCheckList
     $writer = [System.Xml.XmlWriter]::Create($OutputPath.FullName, $xmlWriterSettings)
 
     $writer.WriteStartElement('CHECKLIST')
-
-    #region ASSET
-
     $writer.WriteStartElement("ASSET")
 
+    try
+    {
+        if ($hostName -eq $env:COMPUTERNAME)
+        {
+            [string]$localMac = (Get-NetAdapter | Where-Object {$_.Status -eq "Up"} | Select-Object MacAddress | Select-Object -First 1).MacAddress
+            [string]$localIP = (Get-NetIPAddress -AddressFamily IPV4 | Where-Object { $_.IpAddress -notlike "127.*" } | Select-Object -First 1).IPAddress
+            [string]$localFQDN = [System.Net.DNS]::GetHostByName($env:ComputerName).HostName
+        }
+        else
+        {
+            [string]$localMac = Invoke-Command $hostName -Scriptblock {return (Get-NetAdapter | Where-Object {$_.Status -eq "Up"} | Select-Object MacAddress | Select-Object -First 1).MacAddress}
+            [string]$localIP = Invoke-Command $hostName -Scriptblock {return (Get-NetIPAddress -AddressFamily IPV4 | Where-Object { $_.IpAddress -notlike "127.*" } | Select-Object -First 1).IPAddress}
+            [string]$localFQDN = [System.Net.DNS]::GetHostByName($hostName).HostName
+        }
+    }
+    catch
+    {
+        Write-Warning "Unable to Obtain local IP/MAC Addresses."
+        $localFQDN = $HostName
```

Comment:
using an if to set might be better, furthermore, since you are invoking a command on a remote machine, it makes more sense to query the FQDN directly from the target vs. the source.